src/main/java/com/google/sps/servlets/AuthFilter.java

```diff
@@ -13,68 +13,65 @@
 import java.io.IOException;
-import java.io.FileInputStream;
+import java.util.List;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.annotation.WebInitParam;
 import javax.servlet.annotation.WebServlet;
+import javax.servlet.FilterChain;
+import javax.servlet.Filter;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import org.apache.commons.io.IOUtils;
-import java.io.*;
-import java.util.List;
-import com.google.auth.oauth2.GoogleCredentials;
-import com.google.firebase.auth.FirebaseAuthException;
-import com.google.sps.servlets.MutableHttpServletRequest;
-import com.google.sps.servlets.FirebaseAppInit;
-
 
 @WebFilter(urlPatterns= {"/chatrooms", "/user", "/users", "/getChatroom", "/messages"})
 public class AuthFilter implements Filter {
-
-    FirebaseAppInit firebaseAppInit;
-    FirebaseApp firebaseApp;
+    private FirebaseAppInit firebaseAppInit;
+    private FirebaseApp firebaseApp;
 
     // Create an instance of FirebaseApp.
     public AuthFilter() throws IOException {
         firebaseAppInit = new FirebaseAppInit();
         firebaseApp = firebaseAppInit.initializeFirebaseApp();
     }
 
-    public void init(FilterConfig config) {}
+    public void init(FilterConfig config) throws ServletException {}
 
-    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
-        
-        HttpServletRequest req = (HttpServletRequest) request;
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {  
+        HttpServletRequest servletRequest = (HttpServletRequest) request;
+        HttpServletResponse servletResponse = (HttpServletResponse) response;
+        MutableHttpServletRequest mutableRequest = new MutableHttpServletRequest(servletRequest);
 
         // idToken comes from the Frontend as a parameter.
-        String idToken = req.getHeader("X-token");
-
+        String idToken = servletRequest.getHeader("x-token");
         FirebaseToken decodedUseridToken = null;
         try {
-            decodedUseridToken = FirebaseAuth.getInstance().verifyIdToken(idToken);;
+            decodedUseridToken = FirebaseAuth.getInstance().verifyIdToken(idToken);
         } catch (FirebaseAuthException e) {
             System.out.println(e.getMessage());
         }
 
+        // Attach uid to header if it is not null, else send 401 error message to user.
         String uid = decodedUseridToken.getUid();
-
-        MutableHttpServletRequest mutableRequest = new MutableHttpServletRequest(req);
-
-        mutableRequest.putHeader("userId", uid);
-        chain.doFilter(mutableRequest, response);
+        if(uid != null) {
```

Comment:
Prefer to throw error first

if (uid == null) {
  setError..
  return
}
// Do the rest stuff.

Reply:
okay, noted